### PR TITLE
Add Upsert method to external service store

### LINKED ISSUE #14713: external service store:  Add an Upsert method
```go
func (*ExternalServicesStore) Upsert(ctx context.Context, svcs ...*types.ExternalService) error
```

* Replace all calls to repo-updater's UpsertExternalServices with internal/db/ExternalServicesStore#Upsert
cmd/repo-updater/internal/store.go
* Replace all calls to repo-updater's ListExternalServices with internal/db/ExternalServicesStore#List

Extracted from: https://github.com/sourcegraph/sourcegraph/issues/14092

### PR DESCRIPTION
**Pull request by asdine**
Fixes #14713

### CODE REVIEW

File: cmd/frontend/types/types.go [hunk 1]
```diff
@@ -388,11 +399,555 @@ type ExternalService struct {
...
+// URN returns a unique resource identifier of this external service,
+// used as the key in a repo's Sources map as well as the SourceInfo ID.
 func (e *ExternalService) URN() string {
 	return extsvc.URN(e.Kind, e.ID)
 }
 
+// IsDeleted returns true if the external service is deleted.
+func (e *ExternalService) IsDeleted() bool { return !e.DeletedAt.IsZero() }
+
+// Update updates ExternalService r with the fields from the given newer ExternalService n,
```
Comment (ryanslade) [inline cmd/frontend/types/types.go:411]:
```suggestion
// Update updates ExternalService e with the fields from the given newer ExternalService n,
```

File: cmd/frontend/types/types.go [hunk 2]
```diff
@@ -388,11 +399,555 @@ type ExternalService struct {
...
+
+	if (e.DeletedAt == nil && n.DeletedAt != nil) || (e.DeletedAt != nil && (n.DeletedAt == nil || !e.DeletedAt.Equal(*n.DeletedAt))) {
+		e.DeletedAt, modified = n.DeletedAt, true
+	}
+
+	return modified
+}
+
+// Configuration returns the external service config.
+func (e ExternalService) Configuration() (cfg interface{}, _ error) {
```
Comment (ryanslade) [inline cmd/frontend/types/types.go:442]: I think we should consistently use `e *ExternalService`

**Review by ryanslade — APPROVED** [review]
I assume the plan is to remove all the copied code the `repo-updater` later, do we have a followup issue for that?

**Comment by asdine** [comment]
@ryanslade I'm doing it as part of https://github.com/sourcegraph/sourcegraph/pull/15082

**Pull request merged**